src/server/requestHandlers.js

```diff
@@ -1,431 +1,491 @@
+    gradeNames: "kettle.request.http",
+    invokers: {
+        handleRequest: {
+            funcName: "sjrk.storyTelling.server.handleGetStory",
+            args: ["{request}", "{server}.storyDataSource", "{server}.options.secureConfig.uploadedFilesHandlerPath"]
+        }
+    }
+});
+
+sjrk.storyTelling.server.handleGetStory = function (request, dataSource, uploadedFilesHandlerPath) {
+    var id = request.req.params.id;
+    var promise = dataSource.get({directStoryId: id});
+
+    promise.then(function (response) {
+
+        fluid.transform(response.content, function (block) {
+            if (block.blockType === "image") {
+                if (block.imageUrl) {
+                    block.imageUrl = uploadedFilesHandlerPath + "/" + block.imageUrl;
+                }
+                return block;
+            } else if (block.blockType === "audio" || block.blockType === "video") {
+                if (block.mediaUrl) {
+                    block.mediaUrl = uploadedFilesHandlerPath + "/" + block.mediaUrl;
+                }
+                return block;
+            }
+        });
+
+        var responseAsJSON = JSON.stringify(response);
+        request.events.onSuccess.fire(responseAsJSON);
+    }, function (error) {
+        var errorAsJSON = JSON.stringify(error);
+        request.events.onError.fire({
+            isError: true,
+            message: errorAsJSON
+        });
+    });
+};
+
+fluid.defaults("sjrk.storyTelling.server.saveStoryWithBinariesHandler", {
+    gradeNames: "kettle.request.http",
+    requestMiddleware: {
+        "saveStoryWithBinaries": {
+            middleware: "{server}.saveStoryWithBinaries"
+        }
+    },
+    invokers: {
+        handleRequest: {
+            funcName: "sjrk.storyTelling.server.handleSaveStoryWithBinaries",
+            args: ["{arguments}.0", "{server}.storyDataSource", "{server}.options.globalConfig.authoringEnabled"]
+        }
+    }
+});
+
+sjrk.storyTelling.server.handleSaveStoryWithBinaries = function (request, dataSource, authoringEnabled) {
+    if (authoringEnabled) {
+        var rotateImagePromises = [];
+
+        // rotate any images based on their EXIF data, if present
+        fluid.each(request.req.files.file, function (singleFile) {
+            if (singleFile.mimetype && singleFile.mimetype.indexOf("image") === 0) {
+                rotateImagePromises.push(sjrk.storyTelling.server.rotateImageFromExif(singleFile));
+            }
+        });
+
+        fluid.promise.sequence(rotateImagePromises).then(function () {
+            var storyModel = JSON.parse(request.req.body.model);
+            var binaryRenameMap = sjrk.storyTelling.server.buildBinaryRenameMap(storyModel.content, request.req.files.file);
+
+            sjrk.storyTelling.server.saveStoryToDatabase(dataSource, binaryRenameMap, storyModel, request.events.onSuccess, request.events.onError);
+        }, function (error) {
+            if (!error.errorCode) {

```

Comment:
Why does `request.events.onError` only get fired when `!error.errorCode`? As to line 232-236, `errorCode` could also present when Image rotation fails.